cpp/arcticdb/column_store/column.hpp

```diff
@@ -646,6 +646,72 @@ class Column {
         return *res;
     }
 
+    template<typename input_tdt>
+    static void for_each(const Column& input_column,
+                          std::function<void(typename input_tdt::DataTypeTag::raw_type)>&& f) {
+        auto input_data = input_column.data();
+        std::for_each(input_data.cbegin<input_tdt>(), input_data.cend<input_tdt>(), [&f](auto input_value) {
+            f(input_value);
+        });
+    }
+
+    template<typename input_tdt, typename output_tdt>
+    static void transform(const Column& input_column,
+                          Column& output_column,
+                          std::function<typename output_tdt::DataTypeTag::raw_type(typename input_tdt::DataTypeTag::raw_type)>&& f) {
+        auto input_data = input_column.data();
+        auto output_data = output_column.data();
+        std::transform(input_data.cbegin<input_tdt>(), input_data.cend<input_tdt>(), output_data.begin<output_tdt>(), f);
+    }
+
+    template<typename left_input_tdt, typename right_input_tdt, typename output_tdt>
+    static void transform(const Column& left_input_column,
+                          const Column& right_input_column,
+                          Column& output_column,
+                          std::function<typename output_tdt::DataTypeTag::raw_type(typename left_input_tdt::DataTypeTag::raw_type, typename right_input_tdt::DataTypeTag::raw_type)>&& f) {
+        auto left_input_data = left_input_column.data();
+        auto right_input_data = right_input_column.data();
+        auto output_data = output_column.data();
+        std::transform(left_input_data.cbegin<left_input_tdt>(), left_input_data.cend<left_input_tdt>(), right_input_data.cbegin<right_input_tdt>(), output_data.begin<output_tdt>(), [&f](auto left_value, auto right_value) {
+            return f(left_value, right_value);
+        });
+    }
+
+    template<typename input_tdt>
+    static void transform(const Column& input_column,
+                          util::BitSet& output_bitset,
+                          std::function<bool(typename input_tdt::DataTypeTag::raw_type)>&& f) {
+        auto input_data = input_column.data();
+        util::BitSet::bulk_insert_iterator inserter(output_bitset);
+        auto pos = 0u;
+        std::for_each(input_data.cbegin<input_tdt>(), input_data.cend<input_tdt>(), [&inserter, &pos, &f](auto input_value) {
+            if (f(input_value)) {
+                inserter = pos;
+            }
+            ++pos;
+        });
+        inserter.flush();
+    }
+
+    template<typename left_input_tdt, typename right_input_tdt>
+    static void transform(const Column& left_input_column,
```

Comment:
I think a comment on this would be helpful as it's a bit different to a regular transform (since it takes more parameters)

Reply:
It is equivalent to (3) in https://en.cppreference.com/w/cpp/algorithm/transform